### DISCUSSION TITLE
Zellij exit confirmation

### DISCUSSION THREAD

**Discussion by DumboJetEngine** [zellij-org/zellij#3136] [Ideas]
Hello.

I have been using Zellij for quite some time, and it is my favorite multiplexer (although I've always had font issues in the tab-bar and status-bar areas, that don't bother me much)!

The biggest annoyance I have is this:
Many apps use Ctrl+Q as an exit shortcut and trying to close them, closes Zellij.
I know I can lock the Zellij interface to exit the "inner app", but I sometimes forget to do that.
I also know that I could probably change the Zellij shortcut to something different or try to disable it :  https://github.com/zellij-org/zellij/discussions/3026

But I suggest that you add an exit confirmation dialog to it.
Opening many tabs and panes should be harder to undo than just hitting a single shortcut.
Many people would like to always detach instead of quitting.
Perhaps the Quit section on the status bar should provide two sub-options as a confirmation in the form of a Yes/No or Proceed/Cancel.

**Comment by Lockszmith-GH** [comment]
You can create a quit confirmation right by modifying the configuration yourself. It will not show a dialog, but it will delay the action by asking for another key stroke.

Instead of `Ctrl Q` triggering a quit, make it open a section with another key stroke (maybe a `Y`/`N` one, or requiring another `Ctrl Q`)

**Reply by DumboJetEngine** [reply]
Any quick hints on how to do this?
The list of zellij modes appears to be fixed:
https://zellij.dev/documentation/keybindings-modes
Isn't it?

Can I create a new mode and put zellij into that mode?
Maybe like that? :
```
keybinds {
 unbind "Ctrl q"
 bind "Ctrl q" { SwitchToMode "prequit"; }
 prequit {
  bind "Ctrl q" { Quit; }
 }
}
```
I have no idea if this will work and I haven't tested it.
Hopefully nothing with explode if I use it...

**Reply by Lockszmith-GH** [reply]
I can't test right now, but if it 'explodes' all you need to do is to delete the config.kdl file :)

Looks about right, I would start by [dumping the default](https://zellij.dev/documentation/configuration.html?highlight=dump#how-to-dump-the-default-configuration-to-stdout) and modifying it, instead of starting from scratch.

**Reply by DumboJetEngine** [reply]
It doesn't seem to like it much:
![image](https://github.com/zellij-org/zellij/assets/8676903/05d96049-8c0c-48f3-891e-f1cc3c35d9f8)
I might be able to put the Quit action inside the Session mode instead.

**Reply by DumboJetEngine** [reply]
Yep!
Reusing the `session` mode and adding the Quit inside there worked!
Dandy! ;)
